src/main/java/training/endava/skeleton/Main.java

```diff
@@ -56,17 +60,131 @@ public static void main(String argv[]) {
 
         //Generics demo for lower-bound
 //        YellowFruits y = new YellowFruits();
+        ArrayList<Company> companies = (ArrayList<Company>) makeCompanies(Optional.empty());
+        companies = (ArrayList<Company>) populateCompanies(companies);
+
+        //Stream Exercises
+        //ex 1
+        companies.stream().forEach(System.out::println);
+
+        //ex2
+        //in person implementation (setId)
+
+        //ex3
+//        System.out.println(companies.stream().min((a,b)-> a.getFoundationYear().compareTo(b.getFoundationYear())).get().getName());
+
+        //ex4
+//        System.out.println(companies.stream().map(x -> x.getEmployeeList().size()).reduce(0,(a,b)->a+b));
+
+        //ex5
+//        String name = companies.stream()
+//                .map(x -> x.getEmployeeList()
+//                        .stream()
+//                        .map(y -> y.getName())
+//                        .reduce("", (a, b) -> a + b))
+//                .reduce("", (a, b) -> a + b);
+//        System.out.println(new Person(name, "email", "tel", 123456, companies.get(3)));
```

Comment:
1. Please use flatMap instead of creating a stream inside a stream...
2. It is preferred to use method reference instead. Replace lambdas everywhere is applicable.
3. The idea of the task was to create a Person instance that has the name created as a concatenation, not just a String. Try the reduce method with new Person("") as identity value (constructor for the name attribute) and defining the rule of combining two person instances in the second parameter of the reduce method.